validator/main.py

```diff
@@ -273,64 +290,62 @@ def get_query_function(self, metadata: Dict[str, Any]) -> Callable:
-        #             "`query_function` will be used."
-        #         )
-        #     return query_fn
+        # Check that query_fn or sources are provided
+        if query_fn is not None:
+            if sources is not None:
+                warnings.warn(
+                    "Both `query_function` and `sources` are provided in metadata. "
+                    "`query_function` will be used."
+                )
+            return query_fn
 
-        # if sources is None:
-        #     raise ValueError(
-        #         "You must provide either `query_function` or `sources` in metadata."
-        #     )
+        if sources is None:
+            raise ValueError(
+                "You must provide either `query_function` or `sources` in metadata."
+            )
 
-        # # Check chunking strategy, size and overlap
-        # chunk_strategy = metadata.get("chunk_strategy", "sentence")
-        # if chunk_strategy not in ["sentence", "word", "char", "token"]:
-        #     raise ValueError(
-        #         "`chunk_strategy` must be one of 'sentence', 'word', "
-        #         "'char', or 'token'."
-        #     )
-        # chunk_size = metadata.get("chunk_size", 5)
-        # chunk_overlap = metadata.get("chunk_overlap", 2)
-
-        # # Check embed model
-        # embed_function = metadata.get("embed_function", None)
-        # if embed_function is None:
-        #     # Load model for embedding function
-        #     MODEL = SentenceTransformer("paraphrase-MiniLM-L6-v2")
-
-        #     # Create embed function
-        #     def st_embed_function(sources: list[str]):
-        #         return MODEL.encode(sources)
-
-        #     embed_function = st_embed_function
-
-        def embed_function(text, model="text-embedding-3-small"):
-            text = text[-1]
-            text = text.replace("\n", " ")
-            result = np.array(
-                (
-                    self.embedding_client.embeddings.create(input=[text], model=model)
-                    .data[0]
-                    .embedding
-                )
+        # Check chunking strategy, size and overlap
+        chunk_strategy = metadata.get("chunk_strategy", "sentence")
+        if chunk_strategy not in ["sentence", "word", "char", "token"]:
+            raise ValueError(
+                "`chunk_strategy` must be one of 'sentence', 'word', "
+                "'char', or 'token'."
             )
-            return result
+        chunk_size = metadata.get("chunk_size", 5)
+        chunk_overlap = metadata.get("chunk_overlap", 2)
+
+        # Check embed model
+        embed_function = metadata.get("embed_function", None)
+        if embed_function is None:
+            # Create embed function
+            def st_embed_function(sources: list[str]):
+                print("Running st_embed_function...")
+                return DefaultEncodingModel().encode(sources)
+
+            embed_function = st_embed_function
```

Comment:
We probably don't need `st_embed_function` anymore as a wrapper.  We should be able to just pass `DefaultEncodingModel().encode`